src/styles/blocks/calendar.scss

```
@@ -0,0 +1,47 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap;
+  width: getWidth($cell-size, $gap, $padding);
+  margin: 0 auto;
+  padding: 20px $padding;
+
+  @for $index from 28 through 31 {
```

```suggestion
  @for $index from 28 through $month-days {
```

28 can also be a variable.

---

if I apply the variables in the loop then the loop doesn't work because of the encapsulation too.

---

You can get loop conditions from variables. In your case, only the `$index` variable is changing inside the loop.

https://codepen.io/sergii-nosachenko/pen/gOjNbqG